### ISSUE TITLE
truffle migrate error

### ISSUE THREAD

**Issue by krosza** [DecentLabs/ucd-poc#32]
$ truffle migrate --reset
Compiling .\contracts\ERC20Impl.sol...
Compiling .\contracts\EthBackedLoan.sol...
Compiling .\contracts\Exchange.sol...
Compiling .\contracts\LoanManager.sol...
Compiling .\contracts\OrdersLib.sol...
Compiling .\contracts\Owned.sol...
Compiling .\contracts\Rates.sol...
Compiling .\contracts\TokenUcd.sol...

Compilation warnings encountered:

/C/Users/Iroda/desktop/ucd-poc/contracts/OrdersLib.sol:20:9: Warning: Variable i s declared as a storage pointer. Use an explicit "storage" keyword to silence th is warning.
        Order[] orders;
        ^------------^
,/C/Users/Iroda/desktop/ucd-poc/contracts/OrdersLib.sol:36:9: Warning: Variable  is declared as a storage pointer. Use an explicit "storage" keyword to silence t his warning.
        OrderData order;
        ^-------------^

Writing artifacts to .\build\contracts

Using network 'development'.

Running migration: 1_initial_migration.js
  Replacing Migrations...
  ... undefined
Error encountered, bailing. Network state unknown. Review successful transaction s manually.
Error: Error: Exceeds block gas limit
    at StateManager.queueTransaction (C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\ node_modules\ethereumjs-testrpc\build\cli.node.js:102477:21)
    at GethApiDouble.eth_sendTransaction (C:\Users\Iroda\AppData\Roaming\nvm\v8. 2.1\node_modules\ethereumjs-testrpc\build\cli.node.js:102066:14)
    at GethApiDouble.handleRequest (C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\no de_modules\ethereumjs-testrpc\build\cli.node.js:101882:10)
    at next (C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\node_modules\ethereumjs-t estrpc\build\cli.node.js:52856:18)
    at VmSubprovider.handleRequest (C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\no de_modules\ethereumjs-testrpc\build\cli.node.js:59803:12)
    at next (C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\node_modules\ethereumjs-t estrpc\build\cli.node.js:52856:18)
    at GethDefaults.handleRequest (C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\nod e_modules\ethereumjs-testrpc\build\cli.node.js:101787:12)
    at next (C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\node_modules\ethereumjs-t estrpc\build\cli.node.js:52856:18)
    at FilterSubprovider.handleRequest (C:\Users\Iroda\AppData\Roaming\nvm\v8.2. 1\node_modules\ethereumjs-testrpc\build\cli.node.js:59362:7)
    at next (C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\node_modules\ethereumjs-t estrpc\build\cli.node.js:52856:18)
    at Object.InvalidResponse (C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\node_mo dules\truffle\build\cli.bundled.js:37295:16)
    at C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\node_modules\truffle\build\cli. bundled.js:224765:36
    at C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\node_modules\truffle\build\cli. bundled.js:66971:11
    at C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\node_modules\truffle\build\cli. bundled.js:208348:9
    at XMLHttpRequest.request.onreadystatechange (C:\Users\Iroda\AppData\Roaming \nvm\v8.2.1\node_modules\truffle\build\cli.bundled.js:209773:13)
    at XMLHttpRequestEventTarget.dispatchEvent (C:\Users\Iroda\AppData\Roaming\n vm\v8.2.1\node_modules\truffle\build\cli.bundled.js:67130:18)
    at XMLHttpRequest._setReadyState (C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\ node_modules\truffle\build\cli.bundled.js:67420:12)
    at XMLHttpRequest._onHttpResponseEnd (C:\Users\Iroda\AppData\Roaming\nvm\v8. 2.1\node_modules\truffle\build\cli.bundled.js:67575:12)
    at IncomingMessage.<anonymous> (C:\Users\Iroda\AppData\Roaming\nvm\v8.2.1\no de_modules\truffle\build\cli.bundled.js:67535:24)
    at emitNone (events.js:110:20)

**Comment by szerintedmi** [comment]
ok, hopefully this part fixed.
This might 
```
npm upgrdade truffle -g
git pull
git submodule update --recursive --remote
cd testrpc
npm install
cd ..
```
Note the the multi spending bug in testrpc (see #18 and [testrpc issue](https://github.com/ethereumjs/testrpc/issues/122)) is still not fixed so a forked version must be used for now. 
Use:
```
./runtestrpc2.bat
```
than the usual: `truffle migrate --reset` and `yarn start`

**Comment by szerintedmi** [comment]
resolved now:
1. `runtestrpc2.bat` was incorrect fixed in 775d8d79ebc95690ee9aa8a041320332501dfbf9  
1. forked testrpc submodule build failed on windows b/c of phyton missing.
[installing phyton](https://stackoverflow.com/a/39648550/4006001) solved:
```
npm install --global --production windows-build-tools
npm install --global node-gyp
setx PYTHON "%USERPROFILE%\.windows-build-tools\python27\python.exe"
```

restart cmd prompt, then
```
cd testrpc
npm install
cd ..
./runtestrpc2.bat
```

**Comment by szerintedmi** [comment]
as testrpc is fixed now, forked testrpc (building it and phython install) is not required anymore. Just regular `npm -g upgrade ethereumjs-testrpc`
